client/go/outline/config/config_iptable.go

```diff
@@ -0,0 +1,112 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/netip"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/configyaml"
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/iptable"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type ipTableRootConfig struct {
+	Table []ipTableEntryConfig `yaml:"table"`
+}
+
+type ipTableEntryConfig struct {
+	IP     string                `yaml:"ip,omitempty"`
+	Dialer configyaml.ConfigNode `yaml:"dialer"`
+}
+
+func parseIPTableStreamDialer(
+	ctx context.Context,
+	configMap map[string]any,
+	parseSD configyaml.ParseFunc[*Dialer[transport.StreamConn]],
+) (*Dialer[transport.StreamConn], error) {
+	var rootCfg ipTableRootConfig
+	if err := configyaml.MapToAny(configMap, &rootCfg); err != nil {
+		return nil, fmt.Errorf("failed to map iptable stream config: %w", err)
+	}
+
+	if len(rootCfg.Table) == 0 {
+		return nil, errors.New("iptable config 'table' must not be empty for stream dialer")
+	}
+
+	dialerTable := iptable.NewIPTable[transport.StreamDialer]()
+	allConnTunnelled := true
+	allConnDirect := true
+	for i, entryCfg := range rootCfg.Table {
+		parsedSubDialer, err := parseSD(ctx, entryCfg.Dialer)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse nested stream dialer for table entry %d (ip: %s): %w", i, entryCfg.IP, err)
+		}
+
+		if parsedSubDialer.ConnType == ConnTypeDirect || parsedSubDialer.ConnType == ConnTypePartial {
+			allConnTunnelled = false
+		}
+
+		if parsedSubDialer.ConnType == ConnTypeTunneled || parsedSubDialer.ConnType == ConnTypePartial {
+			allConnDirect = false
+		}
+
+		var currentPrefix netip.Prefix
+		parsedPrefix, errPrefix := netip.ParsePrefix(entryCfg.IP)
+		if errPrefix == nil {
+			currentPrefix = parsedPrefix
+		} else {
+			addr, errAddr := netip.ParseAddr(entryCfg.IP)
+			if errAddr != nil {
+				return nil, fmt.Errorf("iptable entry %d IP '%s' is not a valid IP address or CIDR prefix: failed to parse as prefix (%v) and failed to parse as address (%v)", i, entryCfg.IP, errPrefix, errAddr)
+			}
+			currentPrefix = netip.PrefixFrom(addr, addr.BitLen())
+		}
+
+		dialerTable.AddPrefix(currentPrefix, transport.FuncStreamDialer(parsedSubDialer.Dial))
+	}
+
+	dialer, err := iptable.NewStreamDialer(dialerTable)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create IPTableStreamDialer: %w", err)
+	}
+
+	var connType ConnType
+	if allConnDirect && allConnTunnelled {
+		return nil, fmt.Errorf("internal error: allConnDirect and allConnTunnelled cannot both be true")
```

Comment:
```suggestion
		// This should never happen because we require len(rootCfg.Table) != 0
		return nil, fmt.Errorf("allConnDirect and allConnTunnelled cannot both be true")
```